packages/dumb/src/utils/trends/trends.util.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { range } from 'lodash'
+import { HOURS_IN_DAY, ONE_HOUR_MS } from '../datetime/datetime.util'
+
+const SECTION_INTERVAL_HOURS = 3
+const SECTION_INTERVAL_MS = ONE_HOUR_MS * SECTION_INTERVAL_HOURS
+
+export const getTrendsIntervalsArray = (withInitialValue: boolean = false, withFinalValue: boolean = false): number[] => {
```

Comment:
I feel like this method looks difficult and could be more explicit. Here is my suggestion (all the magic numbers should be in constant oc):

```
export const getTrendsIntervalsArray = (withInitialValue: boolean = false, withFinalValue: boolean = false): number[] => {
  const defaultArray = [
    10800000,
    21600000,
    32400000,
    43200000,
    54000000,
    64800000,
    75600000
  ]
  if (withFinalValue && withInitialValue) {
    return [0, ...defaultArray, 86400000]
  }
  if (withInitialValue) {
    return [0, ...defaultArray]
  }
  if (withFinalValue) {
    return [...defaultArray, 86400000]
  }
  return defaultArray
}
```

edit: I could also see it as this not being a function, but simply 4 constants...